drivers/dma/dma_stm32.c

```diff
@@ -608,8 +608,8 @@ DMA_STM32_EXPORT_API int dma_stm32_stop(const struct device *dev, uint32_t id)
 		return -EINVAL;
 	}
 
-	/* Repeated stop : return now if channel is already stopped */
-	if (!stm32_dma_is_enabled_stream(dma, id)) {
+	/* Repeated stop : return now if channel is already stopped (if not overridden by the HAL)*/
+	if (!stream->hal_override && !stm32_dma_is_enabled_stream(dma, id)) {
 		return 0;
 	}
```

Comment:
In case of `hal_override`, the only thing you need to do is to clear the busy flag.
So I'd suggest to do it directly before anything else and return directly.

Reply:
Note also that same should be done in dma_stm32u5 and dma_stm32_bdma drivers

Reply:
Fixed it in dma_stm32, applied the same change to the other two drivers.